### PR TITLE
sync_diff_inspector: support check between clusters with different time_zone setting

### PR DESCRIPTION
**Pull request by glorv**
### What problem does this PR solve? 
Let sync_diff_inspector support compare between cluster with different default time_zone

### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

Code changes

Side effects

Related changes

### CODE REVIEW

File: pkg/dbutil/common.go [hunk 1]
```diff
@@ -118,6 +119,10 @@ func OpenDB(cfg DBConfig) (*sql.DB, error) {
 		dbDSN = fmt.Sprintf("%s:%s@tcp(%s:%d)/?charset=utf8mb4", cfg.User, cfg.Password, cfg.Host, cfg.Port)
 	}
 
+	for key, val := range vars {
+		dbDSN += fmt.Sprintf("&%s='%s'", key, url.QueryEscape(val))
```
Comment (lance6716) [inline pkg/dbutil/common.go]: seems whole value including `'` should be URLescaped 

https://github.com/go-sql-driver/mysql#system-variables

File: sync_diff_inspector/diff.go [hunk 2]
```diff
@@ -122,21 +123,44 @@ func (df *Diff) init(cfg *Config) (err error) {
...
+		"time_zone": url.QueryEscape(dbutil.FormatTimeZoneOffset(targetTZOffset)),
+	}
+
+	for _, source := range cfg.SourceDBCfg {
+		source.Conn, err = diff.CreateDB(df.ctx, source.DBConfig, nil, cfg.CheckThreadCount)
+		if err != nil {
+			return errors.Annotatef(err, "create source db %s failed", source.DBConfig.String())
+		}
+
+		sourceTZOffset, err := dbutil.GetTimeZoneOffset(df.ctx, cfg.TargetDBCfg.Conn)
```
Comment (lance6716) [inline sync_diff_inspector/diff.go]:
```suggestion
		sourceTZOffset, err := dbutil.GetTimeZoneOffset(df.ctx, source.Conn)
```

And we could directly specify timezone, not try-and-adjust? 🤔

File: tests/sync_diff_inspector/time_zone/run.sh [hunk 3]
```diff
@@ -0,0 +1,33 @@
+
```
Comment (lance6716) [inline tests/sync_diff_inspector/time_zone/run.sh]:
```suggestion
```

File: tests/sync_diff_inspector/time_zone/run.sh [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+mysql -uroot -h 127.0.0.1 -P 4000 -e "create table tz_test.diff(id int, dt datetime, ts timestamp);"
+mysql -uroot -h 127.0.0.1 -P 4000 -e "insert into table tz_test.diff values (1, now(), now());"
+
+echo "dump data and then load to tidb"
+rm -rf $OUT_DIR/dump_tz_diff
+mydumper --host 127.0.0.1 --port 4000 --user root --outputdir $OUT_DIR/dump_tz_diff -B tz_test -T diff
+loader -h 127.0.0.1 -P 4001 -u root -d $OUT_DIR/dump_tz_diff
+
+echo "check with the same time_zone, check result should be pass"
+sync_diff_inspector --config=./config.toml > $OUT_DIR/ignore_column_diff.log || true
```
Comment (lance6716) [inline tests/sync_diff_inspector/time_zone/run.sh]:
```suggestion
sync_diff_inspector --config=./config.toml > $OUT_DIR/different_timezone_diff.log || true
```

why there's `|| true`

**Comment by lance6716** [comment]
the doc says we should use `zap.Any` not `zap.Reflect`, which has better performance

**Comment by glorv** [comment]
@lance6716 @lichunzhu PTAL

**Review by lance6716 — APPROVED** [review]
why we need `|| true`

File: pkg/dbutil/common.go [hunk 5]
```diff
@@ -118,6 +119,10 @@ func OpenDB(cfg DBConfig) (*sql.DB, error) {
 		dbDSN = fmt.Sprintf("%s:%s@tcp(%s:%d)/?charset=utf8mb4", cfg.User, cfg.Password, cfg.Host, cfg.Port)
 	}
 
+	for key, val := range vars {
+		dbDSN += fmt.Sprintf("&%s=%%27%s%%27", key, url.QueryEscape(val))
```
Comment (lichunzhu) [inline pkg/dbutil/common.go:124]: Can you add a comment on this to help others understand or maybe we can wrap quotations before escape?

Comment (glorv) [reply pkg/dbutil/common.go:124]: Done

**Review by lichunzhu — APPROVED** [review]
/lgtm

**Pull request merged**